NewDawn/NewDawn/ProfileViewController/Profile_DraftFinal.swift

```diff
@@ -43,25 +47,28 @@ class Profile_DraftFinal: UIViewController {
             
             if (error != nil){
                 print("noooooo good registration")
+                return
             }
             
             if let register_response = jsonResponse{
                 print("Register Success")
-                self.storeCertification(register_response: register_response)
-                self.notificationSetUp()
-                
-                if let images = ImageUtil.getPersonalImagesWithData(){
-                    for single_image in images{
-                        let single_img = single_image["img"]
-                        let single_params = [
-                            "order": single_image["order"]!,
-                            "caption": single_image["caption"]!,
-                            "user": single_image["user_uri"]!
-                            ] as [String: Any]
-                        let img_name = self.MD5(String(single_image["user_id"] as! Int) + String(single_image["order"] as! Int))! + ".jpeg"
-                        self.photoUploader(photo: single_img as! UIImage, filename: img_name, parameters: single_params){ success in
-                            print("image upload \(success)")}
+                DispatchQueue.main.async {
+                    self.storeCertification(register_response: register_response)
+                    self.notificationSetUp()
+                    if let images = ImageUtil.getPersonalImagesWithData(){
+                        for single_image in images{
+                            let single_img = single_image["img"]
+                            let single_params = [
+                                "order": single_image["order"]!,
+                                "caption": single_image["caption"]!,
+                                "user": single_image["user_uri"]!
+                                ] as [String: Any]
+                            let img_name = self.MD5(String(single_image["user_id"] as! Int) + String(single_image["order"] as! Int))! + ".jpeg"
+                            self.photoUploader(photo: single_img as! UIImage, filename: img_name, parameters: single_params){ success in
+                                print("image upload \(success)")}
+                        }
                     }
+                    self.performSegue(withIdentifier: "after_register", sender: self)
```

Comment:
Do you need DispatchQueue.main.async here?

Reply:
I need to wrap main thread around it. Otherwise it would not work.

Reply:
Does it work if you only wrap it around performSegue instead of whole image upload logic?